### PR TITLE
e2e: Reduce CircleCI credit usage

### PR DESCRIPTION
**Pull request by HowardBraham**
## **Description**

**_TL;DR:_** Reduces credit usage per successful workflow from ~12k to ~9k, see the screenshots and click the links to see for yourself.

### Details
- We were not using the resource class `small` before, but 29 of our jobs run just fine with 1 CPU and 2 GB RAM, most not even taking any extra time
- I moved all resource class `large` back to `medium+`.  This **_might_** cause Out of Memory errors to come back, but I've run this a few times and haven't seen any.
- As @FrederikBolding suggested, we probably don't need to run the Desktop jobs every time.  For now, I've cut it back to only running on the `develop` branch.
- Reduced parallelism on a bunch of jobs, as we were seeing diminishing returns
- Do not run the workflow at all on the branch `l10n_crowdin_action`

### **Before**

[~12k credit usage per workflow on develop, click to see Insights](https://app.circleci.com/insights/github/MetaMask/metamask-extension/workflows/test_and_release/overview?branch=develop&reporting-window=last-24-hours)
image

### **After**

[~9k credit usage per workflow on this branch, click to see Insights](https://app.circleci.com/insights/github/MetaMask/metamask-extension/workflows/test_and_release/overview?branch=reduce-circleci-spend&reporting-window=last-24-hours)
image

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -78,7 +81,10 @@ workflows:
       - trigger-beta-build:
           requires:
             - prep-deps
-      - prep-deps
+      - prep-deps:
+          filters:
+            branches:
+              ignore: l10n_crowdin_action
```
Comment (Gudahtt) [inline .circleci/config.yml:87]: Hmm. Fair point that most of the runs on this branch aren't productive, but in theory we should be regularly merging this branch. And we'd want to ensure it passes CI before merging (we have seen CI catch errors caused by localization problems)

Maybe better to undo this change for now? We're introducing a new translations process soon that should eliminate this branch anyway, we can survive until then.

Comment (HowardBraham) [reply .circleci/config.yml]: Okay undone

**Review by Gudahtt — APPROVED** [review]
LGTM, great work!

**Comment by Gudahtt** [comment]
I'm surprised that any of our e2e test suites work on small instances! If we can keep these machine size reductions, this will make an enormous difference.

**Review by brad-decker — APPROVED** [review]
